src/gg.regression.unity.bots/Runtime/Scripts/CodeCoverage/RGCodeCoverage.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using UnityEngine;
+
+namespace RegressionGames.CodeCoverage
+{
+    /// <summary>
+    /// Class responsible for tracking code coverage.
+    /// </summary>
+    public static class RGCodeCoverage
+    {
+        public static readonly string CodeCovMetadataJsonPath = "Assets/Resources/RGCodeCoverageMetadata.txt";
+        
+        /// <summary>
+        /// Dictionary tracking code coverage in the assemblies.
+        /// Maps assembly name to visited code points.
+        /// This dictionary should always be locked whenever accessing to ensure thread-safety.
+        /// </summary>
+        private static Dictionary<string, ISet<int>> _codeCoverage { get; set; } = new Dictionary<string, ISet<int>>();
+        
+        private static bool _isRecording = false;
+        private static CodeCoverageMetadata _metadata = null;
+
+        public static CodeCoverageMetadata GetMetadata(bool forceReload = false)
+        {
+            if (_metadata != null && !forceReload)
+            {
+                return _metadata;
+            }
+            string json;
+            #if UNITY_EDITOR
+            if (File.Exists(CodeCovMetadataJsonPath))
+            {
+                json = File.ReadAllText(CodeCovMetadataJsonPath);
+            }
+            else
+            {
+                json = null;
+            }
+            #else
+            TextAsset jsonFile = Resources.Load<TextAsset>("RGCodeCoverageMetadata");
+            json = jsonFile?.text;
+            #endif
+            if (json != null)
+            {
+                _metadata = JsonConvert.DeserializeObject<CodeCoverageMetadata>(json);
+            }
+            if (_metadata != null && !_metadata.IsValid())
+            {
+                _metadata = null;
+            }
+            return _metadata;
+        }
+
+        #if UNITY_EDITOR
+        public static void SaveMetadata(CodeCoverageMetadata metadata)
+        {
+            string targetDir = Path.GetDirectoryName(CodeCovMetadataJsonPath);
```

Comment:
Note that the metadata is generated only once when the assemblies are instrumented, so this method will only ever get called from the Unity Editor. This is also the reason why I saved it to the `Resources` folder, so that it gets bundled together with the standalone builds and can continue to be used in production runtimes.